crates/contract/src/update.rs

```diff
@@ -191,6 +192,30 @@ impl ProposedUpdates {
         }
     }
 
+    /// Removes votes from the specified accounts.
+    pub fn remove_votes(&mut self, accounts_to_remove: &[AccountId]) {
+        accounts_to_remove
+            .iter()
+            .for_each(|account| self.remove_vote(account));
+    }
+
+    /// Removes votes from accounts that are not participants.
+    pub fn remove_non_participant_votes(&mut self, participants: &Participants) {
+        let non_participants: Vec<AccountId> = self
+            .vote_by_participant
+            .keys()
+            .filter(|voter| !participants.is_participant(voter))
+            .cloned()
+            .collect();
+
+        self.remove_votes(&non_participants);
+    }
```

Comment:
nit: I would inline this in in `lib.rs` - it's one less method to test and maintain.